### PR TITLE
[somfytahoma] Improved compatibility with the Connexoon box

### PR DESCRIPTION
**Pull request by octa22**
Fixes problem with refreshing values for Connexoon users addressed in the [forum](https://community.openhab.org/t/tahoma-binding-compatible-with-oh2/27004/209?u=ondrej_pecta)

### CODE REVIEW

File: addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaBridgeHandler.java [hunk 1]
```diff
@@ -765,4 +755,7 @@ public String getTahomaStatus(String gatewayId) {
         return null;
     }
 
+    private boolean isAuthenticationChallenge(String message) {
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaBridgeHandler.java]: Would it make sense to let this message accept an exception, now every call contains repetition

File: addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaGatewayHandler.java [hunk 2]
```diff
@@ -50,14 +48,11 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
     public void updateChannelState(ChannelUID channelUID) {
-        if (isStatusExpired()) {
-            String id = getThing().getConfiguration().get("id").toString();
-            if (STATUS.equals(channelUID.getId())) {
-                updateState(channelUID, new StringType(getTahomaStatus(id)));
-                //update the firmware property
-                updateProperty(PROPERTY_FIRMWARE_VERSION, getTahomaVersion(id));
-            }
-            lastUpdated = LocalTime.now();
+        String id = getThing().getConfiguration().get("id").toString();
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaGatewayHandler.java]: This call can be moved inside the if

File: addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/SomfyTahomaBindingConstants.java [hunk 3]
```diff
@@ -148,7 +148,7 @@
     public static final String TAHOMA_URL = API_URL + "externalAPI/json/";
     public static final String TAHOMA_EVENT_URL = API_URL + "enduserAPI/events/";
     public static final String SETUP_URL = API_URL + "enduserAPI/setup/gateways/";
-    public static final String REFRESH_URL = API_URL + "enduserAPI/setup/devices/states/refresh";
+    public static final String REFRESH_URL = TAHOMA_URL + "refreshAllStates";
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/SomfyTahomaBindingConstants.java]: Does this also work for existing devices?

File: addons/binding/org.openhab.binding.somfytahoma/ESH-INF/thing/exteriorscreen.xml [hunk 4]
```diff
@@ -9,7 +9,6 @@
 		</supported-bridge-type-refs>
 		<label>Somfy Exterior Screen</label>
 		<channels>
-			<channel id="position" typeId="position"></channel>
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.somfytahoma/ESH-INF/thing/exteriorscreen.xml]: Is this channel supported / used in the past, does this installations, should we mention it in release notes?

File: addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaBridgeHandler.java (hunk 1 above)
Comment (octa22) [reply addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaBridgeHandler.java]: Done

File: addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/SomfyTahomaBindingConstants.java (hunk 3 above)
Comment (octa22) [reply addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/SomfyTahomaBindingConstants.java]: Yes, it works for both gateways Tahoma and Connexoon.

File: addons/binding/org.openhab.binding.somfytahoma/ESH-INF/thing/exteriorscreen.xml (hunk 4 above)
Comment (octa22) [reply addons/binding/org.openhab.binding.somfytahoma/ESH-INF/thing/exteriorscreen.xml]: This is a relic from the former review, the implementation has been removed but channel not.

File: addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaGatewayHandler.java (hunk 2 above)
Comment (octa22) [reply addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaGatewayHandler.java]: Fixed

**Comment by octa22** [comment]
All comments have been fixed.
Thanks.

**Pull request merged**